ibis/bigquery/compiler.py

```diff
@@ -124,6 +127,76 @@ def _array_length(translator, expr):
     )
 
 
+def _arbitrary(translator, expr):
+    arg, where = expr.op().args
+    if where is not None:
+        arg_formatted = translator.translate(where.ifelse(arg, ibis.NA))
+    else:
+        arg_formatted = translator.translate(arg)
+    return 'ANY_VALUE({})'.format(arg_formatted)
+
+
+def _percentile(translator, expr):
+    args = expr.op().args
+    (arg, percentile_rank) = map(translator.translate, args[:2])
+    value_type = args[2]
+    command = dict(cont='percentile_cont', disc='percentile_disc')[value_type]
+    return '{}({}, {})'.format(command, arg, percentile_rank)
+
+
+def _approx_quantile(translator, expr):
+    args = expr.op().args
+    args = [translator.translate(arg) for arg in args[:2]]
+    # distinct?
+    return 'APPROX_QUANTILES({}, {})'.format(*args)
+
+
+def approx_quantile(arg, number, distinct=False, ignore_nulls=True):
+    if distinct or not ignore_nulls:
+        raise NotImplementedError()
+
+    return bq_ops.ApproxQuantile(arg, number, distinct, ignore_nulls).to_expr()
+
+
+def _approx_count_distinct(translator, expr):
+    arg = expr.op().args[0]
+    arg_formatted = translator.translate(arg)
+    return 'APPROX_COUNT_DISTINCT({})'.format(arg_formatted)
+
+
+def approx_count_distinct(arg):
+    return bq_ops.ApproxCountDistinct(arg).to_expr()
+
+
+def _format_date(translator, expr):
+    (arg, fmt) = map(translator.translate, expr.op().args)
+    return 'FORMAT_DATE({}, {})'.format(fmt, arg)
+
+
+def format_date(fmt, arg):
+    return bq_ops.FormatDate(arg, fmt).to_expr()
+
+
+def _date_diff(translator, expr):
+    (arg0, arg1, date_part) = expr.op().args
+    (af0, af1) = [translator.translate(arg) for arg in (arg0, arg1)]
+    return 'DATE_DIFF({}, {}, {})'.format(af0, af1, date_part)
+
+
+def date_diff(arg0, arg1, date_part='DAY'):
+    return bq_ops.DateDiff(arg0, arg1, date_part).to_expr()
+
+
+def _timestamp_diff(translator, expr):
+    (arg0, arg1, timestamp_part) = expr.op().args
+    (af0, af1) = [translator.translate(arg) for arg in (arg0, arg1)]
+    return 'TIMESTAMP_DIFF({}, {}, {})'.format(af0, af1, timestamp_part)
```

Comment:
We probably want a generic method like `delta` that can choose the correct function to call (`DATE_DIFF`/`TIME_DIFF`/`TIMESTAMP_DIFF`) based on the input types. This would also need a promotion mechanism to determine which type to promote different arguments to.

Reply:
FYI: BigQuery also has a [Datetime type](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime-type) and I don't think we have a way of differentiating in between `Datetime` and `Timestamp` in ibis.